python/python-adaguc-server/ogcapi/routeOGCApi.py

```diff
@@ -0,0 +1,945 @@
+        url, env=adagucenv,  showLogOnError=True)
+
+    # Obtain logfile
+    logfile = adagucInstance.getLogFile()
+    adagucInstance.removeLogFile()
+
+    stage2 = time.perf_counter()
+    logger.info("[PERF] Adaguc execution took: %f", (stage2 - stage1))
+
+    if len(logfile) > 0:
+        logger.info(logfile)
+
+    return status, data
+
+
+def get_extent(coll):
+    """
+    Get the boundingbox extent from the WMS GetCapabilities
+    """
+    if "dataset" in coll:
+        logger.info("callADAGUC by dataset")
+        dataset = coll["dataset"]
+        urlrequest = f"dataset={dataset}&service=wms&version=1.3.0&request=getcapabilities"
+        status, response = callADAGUC(url=urlrequest.encode('UTF-8'))
+        logger.info("status: %d", status)
+        if status == 0:
+            xml = response.getvalue()
+            wms = WebMapService(coll["service"], xml=xml, version='1.3.0')
+        else:
+            logger.error("status: %d", status)
+            return {}
+    else:
+        logger.info("callADAGUC by service")
+        wms = WebMapService(coll["service"], version='1.3.0')
+    return wms.contents[next(iter(wms.contents))].boundingBoxWGS84
+
+
+def get_datasets(adagucDataSetDir):
+    """
+    Return all possible OGCAPI feature datasets, based on the dataset directory
+    """
+    logger.info("getDatasets(%s)", adagucDataSetDir)
+    datasetFiles = [f for f in os.listdir(adagucDataSetDir) if os.path.isfile(
+        os.path.join(adagucDataSetDir, f)) and f.endswith(".xml")]
+    datasets = []
+    for datasetFile in datasetFiles:
+        #logger.info("parsing %s", os.path.join(adagucDataSetDir, datasetFile))
+        try:
+            tree = parse(os.path.join(adagucDataSetDir, datasetFile))
+
+            root = tree.getroot()
+            for ogcapi in root.iter('OgcApiFeatures'):
+                logger.info("ogcapi: %s", ogcapi)
+                dataset = {
+                    "dataset": datasetFile.replace(".xml", ""),
+                    "name": datasetFile.replace(".xml", ""),
+                    "title": datasetFile.replace(".xml", "").lower().capitalize(),
+                    "service": "http://localhost:8080/wms?DATASET="+datasetFile.replace(".xml", ""),
+                }
+                dataset["extent"] = get_extent(dataset)
+                datasets.append(dataset)
+        except ParseError:
+            pass
+
+    return datasets
+
+
+collections = None
+coll_by_name = None
+
+'''
+runs before each request for this blueprint
+'''
+
+
+@routeOGCApi.before_request
+def init_collections():
+    """
+    Calls Generate OGC API Feature collection function before requests starts
+    """
+    generate_collections()
+
+
+def generate_collections():
+    """
+    Generate OGC API Feature collections
+    """
+    # # pylint: disable=global-statement
+    global collections, coll_by_name
+    # TODO: This does not work if dataset configs are changed after the server had been started
```

Comment:
This might some improvement or timer for caching and refreshing. For now disabled it.

Reply:
Solved with cacher solution